pkg/cri/server/sandbox_portforward_windows.go

```diff
@@ -17,61 +17,101 @@
 package server
 
 import (
-	"bytes"
 	"context"
 	"fmt"
 	"io"
+	"net"
+	"time"
 
-	"k8s.io/utils/exec"
-
-	sandboxstore "github.com/containerd/containerd/v2/pkg/cri/store/sandbox"
-	cioutil "github.com/containerd/containerd/v2/pkg/ioutil"
+	"github.com/containerd/log"
+	netutils "k8s.io/utils/net"
 )
 
 func (c *criService) portForward(ctx context.Context, id string, port int32, stream io.ReadWriter) error {
-	stdout := cioutil.NewNopWriteCloser(stream)
-	stderrBuffer := new(bytes.Buffer)
-	stderr := cioutil.NewNopWriteCloser(stderrBuffer)
-	// localhost is resolved to 127.0.0.1 in ipv4, and ::1 in ipv6.
-	// Explicitly using ipv4 IP address in here to avoid flakiness.
-	cmd := []string{"wincat.exe", "127.0.0.1", fmt.Sprint(port)}
-	err := c.execInSandbox(ctx, id, cmd, stream, stdout, stderr)
+	sandbox, err := c.sandboxStore.Get(id)
 	if err != nil {
-		return fmt.Errorf("failed to execute port forward in sandbox: %s: %w", stderrBuffer.String(), err)
+		return fmt.Errorf("failed to find sandbox %q in store: %w", id, err)
 	}
-	return nil
-}
 
-func (c *criService) execInSandbox(ctx context.Context, sandboxID string, cmd []string, stdin io.Reader, stdout, stderr io.WriteCloser) error {
-	// Get sandbox from our sandbox store.
-	sb, err := c.sandboxStore.Get(sandboxID)
-	if err != nil {
-		return fmt.Errorf("failed to find sandbox %q in store: %w", sandboxID, err)
+	var podIP string
+	if !hostNetwork(sandbox.Config) {
+		// get ip address of the sandbox
+		podIP, _, err = c.getIPs(sandbox)
+		if err != nil {
+			return fmt.Errorf("failed to get sandbox ip: %w", err)
+		}
+	} else {
+		// port forwarding is not supported for host process containers
+		return fmt.Errorf("port forwarding not supported for host process containers")
 	}
 
-	// Check the sandbox state
-	state := sb.Status.Get().State
-	if state != sandboxstore.StateReady {
-		return fmt.Errorf("sandbox is in %s state", fmt.Sprint(state))
-	}
+	err = func() error {
+		var conn net.Conn
+		if netutils.IsIPv4String(podIP) {
+			conn, err = net.Dial("tcp4", fmt.Sprintf("%s:%d", podIP, port))
+			if err != nil {
+				return fmt.Errorf("failed to connect to %s:%d inside namespace %q, IPv4: %v", podIP, port, id, err)
```

Comment:
not really in a namespace.. maybe 
```suggestion
				return fmt.Errorf("failed to connect to %s:%d for pod %q: %v", podIP, port, id, err)
```
ipv4 vs 6 should be obvious by the ip being referred to.. :-)